unit_test/test_spdm_requester/get_csr.c

```diff
@@ -158,6 +187,43 @@ void libspdm_test_requester_get_csr_case2(void **state)
     assert_memory_equal(csr_form_get, csr_data_pointer, global_csr_len);
 }
 
+/**
+ * Test 3: Successful response to set certificate for slot 0,
+ * with a reset required
+ * Expected Behavior: get a RETURN_SUCCESS return code
+ **/
+void libspdm_test_requester_get_csr_case3(void **state)
+{
+    libspdm_return_t status;
+    libspdm_test_context_t *spdm_test_context;
+    libspdm_context_t *spdm_context;
+
+    uint8_t csr_form_get[LIBSPDM_MAX_CSR_SIZE] = {0};
+    size_t csr_len;
+
+    csr_len = LIBSPDM_MAX_CSR_SIZE;
+
+    spdm_test_context = *state;
+    spdm_context = spdm_test_context->spdm_context;
+    spdm_test_context->case_id = 0x3;
+    spdm_context->connection_info.version = SPDM_MESSAGE_VERSION_12 <<
+                                            SPDM_VERSION_NUMBER_SHIFT_BIT;
+
+    spdm_context->connection_info.connection_state =
+        LIBSPDM_CONNECTION_STATE_NEGOTIATED;
+    spdm_context->connection_info.capability.flags |=
+        SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_CSR_CAP |
+        SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_CERT_INSTALL_RESET_CAP;
+
+    status = libspdm_get_csr(spdm_context, NULL, NULL, 0, NULL, 0, (void *)&csr_form_get,
+                             &csr_len);
+
+    assert_int_equal(status, LIBSPDM_STATUS_RESET_REQUIRED_PEER);

```

Comment:
`assert_int_equal(status, LIBSPDM_STATUS_RESET_REQUIRED_PEER);` is the only check that is needed.